pkg/resource/factory.go

```diff
@@ -66,7 +68,31 @@ func (rf *Factory) SliceFromPatches(
 		if err != nil {
 			return nil, internal.Handler(err, string(path))
 		}
-		result = append(result, res...)
+		for len(res) > 0 {
```

Comment:
I suggest to put this change into `SliceFromBytes` and leave `SliceFromPatches` untouched. Then both resources and patches will be able to read `List`.